File: apps/node/test/fizzbuzz.test.ts
import { FizzBuzz } from "../src/fizzbuzz";

describe("FizzBuzz", () => {
  let fizzbuzz: FizzBuzz;

  beforeEach(() => {
    fizzbuzz = new FizzBuzz();
  });

  test("1を渡したら文字列1を返す", () => {
    expect(fizzbuzz.generate(1)).toBe("1");
  });

  test("2を渡したら文字列2を返す", () => {
    expect(fizzbuzz.generate(2)).toBe("2");
  });

  test("3を渡したらFizzを返す", () => {
    expect(fizzbuzz.generate(3)).toBe("Fizz");
  });

  test("5を渡したらBuzzを返す", () => {
    expect(fizzbuzz.generate(5)).toBe("Buzz");
  });

  test("15を渡したらFizzBuzzを返す", () => {
    expect(fizzbuzz.generate(15)).toBe("FizzBuzz");
  });

  test("generateList(100) は FizzBuzz の配列を返す", () => {
    const result = fizzbuzz.generateList(100);

    expect(result).toHaveLength(100);
    expect(result[0]).toBe("1");
    expect(result[1]).toBe("2");
    expect(result[2]).toBe("Fizz");
    expect(result[3]).toBe("4");
    expect(result[4]).toBe("Buzz");
    expect(result[14]).toBe("FizzBuzz");
    expect(result[99]).toBe("Buzz");
  });
});

File: apps/elixir/test/fizz_buzz_test.exs
defmodule FizzBuzzTest do
  use ExUnit.Case

  describe "generate/1" do
    test "1 を渡すと文字列 1 を返す" do
      assert FizzBuzz.generate(1) == "1"
    end

    test "2 を渡すと文字列 2 を返す" do
      assert FizzBuzz.generate(2) == "2"
    end

    test "3 の倍数を渡すと Fizz を返す" do
      assert FizzBuzz.generate(3) == "Fizz"
      assert FizzBuzz.generate(6) == "Fizz"
    end

    test "5 の倍数を渡すと Buzz を返す" do
      assert FizzBuzz.generate(5) == "Buzz"
    end

    test "15 の倍数を渡すと FizzBuzz を返す" do
      assert FizzBuzz.generate(15) == "FizzBuzz"
    end
  end

  describe "generate_list/0" do
    test "1 から 100 までの FizzBuzz リストを返す" do
      result = FizzBuzz.generate_list()

      assert length(result) == 100
      assert Enum.at(result, 0) == "1"
      assert Enum.at(result, 2) == "Fizz"
      assert Enum.at(result, 4) == "Buzz"
      assert Enum.at(result, 14) == "FizzBuzz"
    end
  end
end
